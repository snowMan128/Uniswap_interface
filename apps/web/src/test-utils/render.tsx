import { MockedProvider } from '@apollo/client/testing'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { queries } from '@testing-library/dom'
import { RenderOptions, render } from '@testing-library/react'
import { RenderHookOptions, WrapperComponent, renderHook } from '@testing-library/react-hooks'
import Web3Provider from 'components/Web3Provider'
import { AssetActivityProvider } from 'graphql/data/apollo/AssetActivityProvider'
import { TokenBalancesProvider } from 'graphql/data/apollo/TokenBalancesProvider'
import { BlockNumberContext } from 'lib/hooks/useBlockNumber'
import { PropsWithChildren, ReactElement, ReactNode } from 'react'
import { HelmetProvider } from 'react-helmet-async/lib/index'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import store from 'state'
import { ThemeProvider } from 'theme'
import { TamaguiProvider } from 'theme/tamaguiProvider'
import { UnitagUpdaterContextProvider } from 'uniswap/src/features/unitags/context'

const queryClient = new QueryClient()

const BLOCK_NUMBER_CONTEXT = { fastForward: () => {}, block: 1234, mainnetBlock: 1234 }
function MockedBlockNumberProvider({ children }: PropsWithChildren) {
  return <BlockNumberContext.Provider value={BLOCK_NUMBER_CONTEXT}>{children}</BlockNumberContext.Provider>
}

const WithProviders = ({ children }: { children?: ReactNode }) => {
  return (
    <HelmetProvider>
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <BrowserRouter>
            <Web3Provider>
              <MockedProvider showWarnings={false}>
                <AssetActivityProvider>
                  <TokenBalancesProvider>
                    <MockedBlockNumberProvider>
                      <UnitagUpdaterContextProvider>
                        <ThemeProvider>
                          <TamaguiProvider>{children}</TamaguiProvider>
                        </ThemeProvider>
                      </UnitagUpdaterContextProvider>
                    </MockedBlockNumberProvider>
                  </TokenBalancesProvider>
                </AssetActivityProvider>
              </MockedProvider>
            </Web3Provider>
          </BrowserRouter>
        </QueryClientProvider>
      </Provider>
    </HelmetProvider>
  )
}

type CustomRenderOptions = Omit<RenderOptions, 'wrapper'>
const customRender = (ui: ReactElement, options?: CustomRenderOptions) => {
  return render<typeof queries>(ui, { ...options, wrapper: WithProviders })
}

type CustomRenderHookOptions<Props> = Omit<RenderHookOptions<Props>, 'wrapper'>
const customRenderHook = <Result, Props>(
  hook: (initialProps: Props) => Result,
  options?: CustomRenderHookOptions<Props>
) => {
  return renderHook(hook, { ...options, wrapper: WithProviders as WrapperComponent<Props> })
}

// Testing utils may export *.
// eslint-disable-next-line no-restricted-syntax
export * from '@testing-library/react'
export { customRender as render, customRenderHook as renderHook }
