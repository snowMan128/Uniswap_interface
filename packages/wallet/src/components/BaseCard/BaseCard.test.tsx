import { View } from 'react-native'
import { ON_PRESS_EVENT_PAYLOAD } from 'wallet/src/test/eventFixtures'
import { fireEvent, render } from 'wallet/src/test/test-utils'
import { BaseCard } from './BaseCard'

describe(BaseCard.Shadow, () => {
  it('renders without error', () => {
    const tree = render(<BaseCard.Shadow />)
    expect(tree).toMatchSnapshot()
  })
})

describe(BaseCard.Header, () => {
  const title = 'title'
  const subtitle = 'subtitle'

  it('renders without error', () => {
    const tree = render(<BaseCard.Header title={title} />)

    expect(tree).toMatchSnapshot()
  })

  describe('title', () => {
    it('renders title text if string is passed', async () => {
      const { queryByText } = render(<BaseCard.Header title={title} />)

      expect(queryByText(title)).toBeTruthy()
    })

    it('renders custom title component if component is passed', () => {
      const { queryByTestId } = render(<BaseCard.Header title={<View testID="custom-title" />} />)

      expect(queryByTestId('custom-title')).toBeTruthy()
    })
  })

  describe('subtitle', () => {
    it('renders subtitle text if string is passed', () => {
      const { queryByText } = render(<BaseCard.Header subtitle={subtitle} title={title} />)

      expect(queryByText(subtitle)).toBeTruthy()
    })

    it('renders custom subtitle component if component is passed', () => {
      const { queryByTestId } = render(
        <BaseCard.Header subtitle={<View testID="custom-subtitle" />} title={title} />
      )

      expect(queryByTestId('custom-subtitle')).toBeTruthy()
    })
  })

  describe('icon', () => {
    it('renders icon if icon is passed', () => {
      const { queryByTestId } = render(
        <BaseCard.Header icon={<View testID="custom-icon" />} title={title} />
      )

      expect(queryByTestId('custom-icon')).toBeTruthy()
    })
  })

  describe('onPress', () => {
    it('calls onPress if onPress is passed and title is pressed', () => {
      const onPress = jest.fn()
      const { getByTestId } = render(
        <BaseCard.Header testID="header" title={title} onPress={onPress} />
      )

      expect(onPress).toHaveBeenCalledTimes(0)

      fireEvent.press(getByTestId('header'), ON_PRESS_EVENT_PAYLOAD)

      expect(onPress).toHaveBeenCalledTimes(1)
    })
  })
})

describe(BaseCard.EmptyState, () => {
  it('renders without error', () => {
    const tree = render(<BaseCard.EmptyState description="description" />)

    expect(tree).toMatchSnapshot()
  })

  it('renders description', () => {
    const { queryByText } = render(<BaseCard.EmptyState description="description" />)

    expect(queryByText('description')).toBeTruthy()
  })

  it('renders icon when icon is passed', () => {
    const { queryByTestId } = render(
      <BaseCard.EmptyState description="description" icon={<View testID="custom-icon" />} />
    )

    expect(queryByTestId('custom-icon')).toBeTruthy()
  })

  it('renders title when title is passed', () => {
    const { queryByText } = render(<BaseCard.EmptyState description="description" title="title" />)

    expect(queryByText('title')).toBeTruthy()
  })

  describe('button', () => {
    it('renders button when buttonLabel is passed', () => {
      const { queryByText } = render(
        <BaseCard.EmptyState buttonLabel="buttonLabel" description="description" />
      )

      expect(queryByText('buttonLabel')).toBeTruthy()
    })

    it('calls onPress when button is pressed', () => {
      const onPress = jest.fn()
      const { getByText } = render(
        <BaseCard.EmptyState
          buttonLabel="buttonLabel"
          description="description"
          onPress={onPress}
        />
      )

      expect(onPress).toHaveBeenCalledTimes(0)

      fireEvent.press(getByText('buttonLabel'), ON_PRESS_EVENT_PAYLOAD)

      expect(onPress).toHaveBeenCalledTimes(1)
    })
  })

  describe('additional button', () => {
    it('renders additional button when additionalButtonLabel is passed', () => {
      const { queryByText } = render(
        <BaseCard.EmptyState
          additionalButtonLabel="additionalButtonLabel"
          description="description"
        />
      )

      expect(queryByText('additionalButtonLabel')).toBeTruthy()
    })

    it('calls onPressAdditional when additional button is pressed', () => {
      const onPressAdditional = jest.fn()
      const { getByText } = render(
        <BaseCard.EmptyState
          additionalButtonLabel="additionalButtonLabel"
          description="description"
          onPressAdditional={onPressAdditional}
        />
      )

      expect(onPressAdditional).toHaveBeenCalledTimes(0)

      fireEvent.press(getByText('additionalButtonLabel'), ON_PRESS_EVENT_PAYLOAD)

      expect(onPressAdditional).toHaveBeenCalledTimes(1)
    })
  })
})

describe(BaseCard.ErrorState, () => {
  it('renders without error', () => {
    const tree = render(<BaseCard.ErrorState />)

    expect(tree).toMatchSnapshot()
  })

  it('renders description when description is passed', () => {
    const { queryByText } = render(<BaseCard.ErrorState description="description" />)

    expect(queryByText('description')).toBeTruthy()
  })

  it('renders title when title is passed', () => {
    const { queryByText } = render(<BaseCard.ErrorState description="description" title="title" />)

    expect(queryByText('title')).toBeTruthy()
  })

  describe('retry button', () => {
    it('renders retry button when retryButtonLabel is passed', () => {
      const { queryByText } = render(
        <BaseCard.ErrorState description="description" retryButtonLabel="retryButtonLabel" />
      )

      expect(queryByText('retryButtonLabel')).toBeTruthy()
    })

    it('calls onRetry when retry button is pressed', () => {
      const onRetry = jest.fn()
      const { getByText } = render(
        <BaseCard.ErrorState
          description="description"
          retryButtonLabel="retryButtonLabel"
          onRetry={onRetry}
        />
      )

      expect(onRetry).toHaveBeenCalledTimes(0)

      fireEvent.press(getByText('retryButtonLabel'), ON_PRESS_EVENT_PAYLOAD)

      expect(onRetry).toHaveBeenCalledTimes(1)
    })
  })

  it('renders icon when icon is passed', () => {
    const { queryByTestId } = render(
      <BaseCard.ErrorState description="description" icon={<View testID="custom-icon" />} />
    )

    expect(queryByTestId('custom-icon')).toBeTruthy()
  })
})

describe(BaseCard.InlineErrorState, () => {
  it('renders without error', () => {
    const tree = render(<BaseCard.InlineErrorState />)

    expect(tree).toMatchSnapshot()
  })

  describe('title', () => {
    it('renders default title when title is not passed', () => {
      const { queryByText } = render(<BaseCard.InlineErrorState />)

      expect(queryByText('Oops! Something went wrong.')).toBeTruthy()
    })

    it('renders custom title when title is passed', () => {
      const { queryByText } = render(<BaseCard.InlineErrorState title="custom-title" />)

      expect(queryByText('custom-title')).toBeTruthy()
    })
  })

  describe('retry button', () => {
    it('does not render retry button when onRetry is not passed', () => {
      const { queryByText } = render(<BaseCard.InlineErrorState />)

      expect(queryByText('Retry')).toBeFalsy()
    })

    it('renders default retry button when retryButtonLabel is not passed', () => {
      const { queryByText } = render(<BaseCard.InlineErrorState onRetry={jest.fn()} />)

      expect(queryByText('Retry')).toBeTruthy()
    })

    it('renders custom retry button when retryButtonLabel is passed', () => {
      const { queryByText } = render(
        <BaseCard.InlineErrorState retryButtonLabel="custom-label" onRetry={jest.fn()} />
      )

      expect(queryByText('custom-label')).toBeTruthy()
    })

    it('calls onRetry when retry button is pressed', () => {
      const onRetry = jest.fn()
      const { getByText } = render(<BaseCard.InlineErrorState onRetry={onRetry} />)

      expect(onRetry).toHaveBeenCalledTimes(0)

      fireEvent.press(getByText('Retry'), ON_PRESS_EVENT_PAYLOAD)

      expect(onRetry).toHaveBeenCalledTimes(1)
    })
  })

  describe('icon', () => {
    it('renders default icon when icon is not passed', () => {
      const { queryByTestId } = render(<BaseCard.InlineErrorState />)

      expect(queryByTestId('error-icon')).toBeTruthy()
    })

    it('renders custom icon when icon is passed', () => {
      const { queryByTestId } = render(
        <BaseCard.InlineErrorState icon={<View testID="custom-icon" />} />
      )

      expect(queryByTestId('custom-icon')).toBeTruthy()
    })
  })
})
