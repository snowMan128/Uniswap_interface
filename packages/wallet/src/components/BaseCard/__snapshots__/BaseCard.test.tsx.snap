// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState renders without error 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "column",
      "gap": 24,
      "justifyContent": "center",
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "gap": 16,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "column",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={true}
        maxFontSizeMultiplier={1.4}
        style={
          {
            "color": "#7D7D7D",
            "fontFamily": "Basel-Book",
            "fontSize": 17,
            "lineHeight": 24,
            "textAlign": "center",
          }
        }
        suppressHighlighting={true}
      >
        description
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "stretch",
        "flexDirection": "row",
        "gap": 16,
      }
    }
  />
</View>
`;

exports[`ErrorState renders without error 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "column",
      "flexGrow": 1,
      "gap": 24,
      "justifyContent": "center",
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "gap": 16,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "column",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={true}
        maxFontSizeMultiplier={1.4}
        style={
          {
            "color": "#7D7D7D",
            "fontFamily": "Basel-Book",
            "fontSize": 17,
            "lineHeight": 24,
            "textAlign": "center",
          }
        }
        suppressHighlighting={true}
      >
        Something went wrong
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "stretch",
        "flexDirection": "row",
      }
    }
  />
</View>
`;

exports[`Header renders without error 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    {
      "bottom": 5,
      "left": 5,
      "right": 5,
      "top": 5,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderBottomColor": "#2222220D",
      "borderBottomWidth": 0.25,
      "borderStyle": "solid",
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 12,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 16,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "stretch",
          "flexDirection": "column",
          "gap": 4,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          }
        }
      >
        <Text
          allowFontScaling={true}
          maxFontSizeMultiplier={1.4}
          style={
            {
              "color": "#7D7D7D",
              "fontFamily": "Basel-Book",
              "fontSize": 17,
              "lineHeight": 24,
            }
          }
          suppressHighlighting={true}
        >
          title
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "borderBottomLeftRadius": 999999,
          "borderBottomRightRadius": 999999,
          "borderTopLeftRadius": 999999,
          "borderTopRightRadius": 999999,
          "flexDirection": "column",
          "justifyContent": "center",
          "transform": [
            {
              "rotate": "180deg",
            },
          ],
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="20"
        bbWidth="24"
        fill="none"
        focusable={false}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="currentColor"
        strokeWidth={8}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "color": "#7D7D7D",
              "height": 20,
              "width": 24,
            },
            {
              "flex": 0,
              "height": 20,
              "width": 24,
            },
          ]
        }
        tintColor="#7D7D7D"
        vbHeight={24}
        vbWidth={24}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeWidth="8"
        >
          <RNSVGPath
            d="M15 6L9 12L15 18"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="3"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;

exports[`InlineErrorState renders without error 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F9F9F9",
      "borderBottomLeftRadius": 16,
      "borderBottomRightRadius": 16,
      "borderTopLeftRadius": 16,
      "borderTopRightRadius": 16,
      "flexDirection": "row",
      "flexGrow": 1,
      "gap": 24,
      "justifyContent": "space-between",
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "gap": 8,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="16"
      fill="currentColor"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      strokeWidth={8}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "color": "#CECECE",
            "height": 16,
            "width": 16,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      testID="error-icon"
      tintColor="#CECECE"
      vbHeight={24}
      vbWidth={24}
    >
      <RNSVGGroup
        fill={
          {
            "type": 2,
          }
        }
        propList={
          [
            "fill",
            "strokeWidth",
          ]
        }
        strokeWidth="8"
      >
        <RNSVGPath
          d="M21.6074 17.1522L15.0004 4.79599C13.7204 2.40199 10.2814 2.40199 9.00042 4.79599L2.39349 17.1522C1.21949 19.3482 2.81353 22.0001 5.30853 22.0001H18.6923C21.1863 22.0001 22.7814 19.3472 21.6074 17.1522ZM11.2504 10.0001C11.2504 9.58609 11.5864 9.25009 12.0004 9.25009C12.4144 9.25009 12.7504 9.58609 12.7504 10.0001V14.0001C12.7504 14.4141 12.4144 14.7501 12.0004 14.7501C11.5864 14.7501 11.2504 14.4141 11.2504 14.0001V10.0001ZM12.0204 18.0001C11.4684 18.0001 11.0153 17.5521 11.0153 17.0001C11.0153 16.4481 11.4584 16.0001 12.0104 16.0001H12.0204C12.5734 16.0001 13.0204 16.4481 13.0204 17.0001C13.0204 17.5521 12.5724 18.0001 12.0204 18.0001Z"
          fill={
            {
              "type": 2,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.4}
      numberOfLines={1}
      style={
        {
          "color": "#222222",
          "fontFamily": "Basel-Book",
          "fontSize": 17,
          "lineHeight": 24,
          "textAlign": "center",
        }
      }
      suppressHighlighting={true}
    >
      Oops! Something went wrong.
    </Text>
  </View>
</View>
`;

exports[`Shadow renders without error 1`] = `
<View
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "#FFFFFFff",
      "borderBottomLeftRadius": 16,
      "borderBottomRightRadius": 16,
      "borderTopLeftRadius": 16,
      "borderTopRightRadius": 16,
      "flexDirection": "column",
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "shadowColor": "rgb(0,0,0)",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.0075,
      "shadowRadius": 10,
    }
  }
/>
`;
